app/scripts/modules/core/src/managed/artifactDetail/ArtifactDetail.tsx

```diff
@@ -152,22 +121,31 @@ const EnvironmentCards = memo(
         }
       />
     );
-    const versionStateCard = (
-      <VersionStateCard
-        key="versionStateCard"
-        state={state}
-        deployedAt={deployedAt}
-        replacedAt={replacedAt}
-        replacedBy={replacedBy}
-        vetoed={vetoed}
-        compareLink={compareLink}
-        allVersions={allVersions}
-        logClick={(message) => logEvent(message, application.name, environmentName, reference)}
-      />
-    );
-    const constraintCards = useMemo(
-      () =>
-        [...(statelessConstraints || []), ...(statefulConstraints || [])]
+
+    return (
+      <>
+        {differentVersionPinnedCard}
+        {pinnedCard}
+        <VersionStateCard
+          key="versionStateCard"
+          state={state}
+          deployedAt={deployedAt}
+          replacedAt={replacedAt}
+          replacedBy={replacedBy}
+          vetoed={vetoed}
+          compareLink={compareLink}
+          allVersions={allVersions}
+          logClick={(action) => logEvent({ action, label: `${environmentName}:${reference}` })}
+        />
+        {environment.verifications?.map((verification) => (
+          <VerificationCard
+            key={verification.id}
+            {...verification}
```

Comment:
Since `VerificationCard` renders a Verification, why not just pass a single prop `verification={verification}`?  Is there a benefit to using spread here?

My preference would be either to be explicit about what properties of `verification` are being used (`foo={verification.foo} bar={verification.bar}`) or pass the whole object in as a single prop.

Reply:
I don't have a strong preference here, I usually prefer to have flat props when possible. I'll change it